include/filters/filter_chain.hpp

```diff
@@ -466,15 +472,16 @@ class MultiChannelFilterChain
        
     for (int i = 0; i < config.size(); ++i)
     {
-      boost::shared_ptr<filters::MultiChannelFilterBase<T> > p(loader_.createInstance(config[i]["type"]));
-      if (p.get() == NULL)
+      auto p(loader_.createUnmanagedInstance(config[i]["type"]));
```

Comment:
Could you add a comment here explaining what you said in the PR? (that we create an unmanaged plugin first because pluginlib doesn't have a version that returns `std::shared_ptr`.